packages/internal/src/build/babel/common.ts

```diff
@@ -0,0 +1,14 @@
+import type { TransformOptions } from '@babel/core'
+
+export interface RequireHookOptions extends TransformOptions {
+  extensions?: string[]
+  cache?: boolean
+}
+
+// @NOTE
+// We do this so we still get type support, but don't import babel/register
+// Importing babel/register in typescript (instead of requiring) has dire consequences...
```

Comment:
Super curious about this? In TypeScript?

Reply:
I believe its the way babel-preset-typescript (and maybe tsc) converts the import statement. 

Best way to describe it:
Lets say we use the import syntax: `import babelRequireHook from '@babel/register'`

- if your import in a js file (like we use to in the cli project) - wasn't a problem, and it would only invoke the register function when you called `babelRequireHook`
- if you import in a TS file, the transpile process modifies it. So it will invoke it once as soon as you import, and another time when you use `babelRequireHook`...
BUTTT!!! you won't notice it if your project is TS because by default it ignore `.ts` and `.tsx` files, but if its a JS project, it would try to transpile twice

Reply:
Could you add this as a comment - I think others might want to understand the concern later down the line.